app/helpers/blacklight/catalog_helper_behavior.rb

```diff
@@ -78,7 +78,7 @@ def search_field_label(params)
   #
   # @return [Blacklight::Configuration::SortField]
   def current_sort_field
-    blacklight_config.sort_fields[params[:sort]] || default_sort_field
+    (blacklight_config.sort_fields[@response.sort] if @response and @response.sort.present?) || blacklight_config.sort_fields[params[:sort]] || default_sort_field
```

Comment:
I think we need to be trickier here. sort fields are not necessarily keyed on the list of fields. I think we'll need something like e.g.:

```
blacklight_config.sort_fields.find { |f| f.sort == @response.sort }
```